src/renderers/dom/shared/inputValueTracking.js

```diff
@@ -40,26 +30,30 @@ function isCheckable(elem: any) {
   );
 }
 
-function getTracker(inst: any) {
-  if (typeof inst.tag === 'number') {
-    inst = inst.stateNode;
-  }
-  return inst._wrapperState.valueTracker;
+function getTracker(node: ElementWithValueTracker) {
+  return node._valueTracker;
 }
 
-function detachTracker(subject: SubjectWithWrapperState) {
-  subject._wrapperState.valueTracker = null;
+function detachTracker(node: ElementWithValueTracker) {
+  node._valueTracker = null;
 }
 
-function getValueFromNode(node: any) {
-  var value;
-  if (node) {
-    value = isCheckable(node) ? '' + node.checked : node.value;
+function getValueFromNode(node: HTMLInputElement): string {
+  var value = '';
+  if (!node) {
+    return value;
```

Comment:
@gaearon I think it's fine, `getValueFromNode` is just used to compare next and last values, both of which are returned from `getValueFromNode`. So the equality check should still hold.